### PR TITLE
Added the ability to log errors with a logging library.

### PR DESCRIPTION
**Pull request by serenitygrant**

### CODE REVIEW

**Comment by serenitygrant** [comment]
@novaki Can this be merged? I want to test the logging in AWS on EC2.

File: src/Pump.coffee [hunk 1]
```diff
@@ -217,6 +217,16 @@ module.exports = class Pump extends EventEmitter
         console.log "Error in pump #{name}: #{errorRecord.error}"
     @
 
+  logErrorsToLogger: (logger) ->
+    @errorBuffer().on 'write', (errorRecord) =>
+      name = errorRecord.pump ? '(root)'
```
Comment (novaki) [inline src/Pump.coffee]: @serenitygrant Please check if logger is an object has an error method here.

**Comment by serenitygrant** [comment]
@novaki Check is in. It'll default to console if the logger doesn't have that function. I hope to make it more robust for different types on loggers later on as well.

**Comment by novaki** [comment]
@serenitygrant I've updated the method to check logger method on call, and removed logging to console when the logger is not valid. Logging to console instead of reporting the error to the developer may lead to unexpected/difficult to debug situations. Included the logErrorsToLogger method in 0.5.1 release.

**Comment by serenitygrant** [comment]
Makes sense! Thanks!

**Pull request merged**